feature: Add "last seen" info for lines with no trains

idreyn opened the pull request:
So thanks to @mathcolo and his work in #34, I had a good base to work from in implementing "last seen" information for lines that currently have no new trains running. It's not fancy, but it looks like this:

<img src="https://user-images.githubusercontent.com/2208769/99326220-e1bc4a80-2845-11eb-8a8a-e64337ee9f2d.png" height="600px">

To fully configure this in production, we'll need to add

```
npm run create-history-db
```
to our deploy process, and then
```
npm run update-history-db
```
to our crontab or something to run periodically.

 _Test plan:_
- Run `npm run create-history-db`
- Update `npm run update-history-db` so that the update function is called with `test_mode=True` and then run `npm run update-history-db` to populate the database with some info, even if there are currently no new trains
- Tweak `initial_request_data()` to pass `test_mode=True` to `get_recent_sightings_for_lines`
- Open the train tracker and verify that for lines without new trains, you see a message about when a train (not actually new in this case) was actually seen on this line.

mathcolo commented on history/util.py:
```diff
@@ -0,0 +1,22 @@
+
+HISTORY_DB_SCHEMA = """
+CREATE TABLE IF NOT EXISTS newtrains_history (
+    id UUID DEFAULT uuid_generate_v4(),
```
This better than the nextval that's currently there?

idreyn replied:
Not at all. I'm a UUID fiend and I didn't check the existing schema, but I agree that `nextval` is simpler!

mathcolo commented on history/create_history_db.py:
```diff
@@ -0,0 +1,21 @@
+    cxn = get_history_db_connection()
+    cxn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+    cur = cxn.cursor()
+    cur.execute('CREATE EXTENSION IF NOT EXISTS "uuid-ossp";')
```
I assume this is for the id column? See comment below

idreyn replied:
Yep, that's right.

mathcolo commented on server/secrets.py:
```diff
@@ -1,7 +1,7 @@
-'''
+MBTA_V3_API_KEY = ""
+POSTGRES_DB = "newtrains"
+POSTGRES_USER = "ian"  # No password; unix account auth
```
Should this not be ubuntu?

idreyn replied:
Oh, yeah, I thought I configured this to not commit my changes to secrets! (Luckily there are no actual secrets in here)

mathcolo reviewed (approved):
Awesome! few comments, but otherwise seems fine to me!

nathan-weinberg reviewed (changes requested):
Could we add the new `history/` directory to the linting check as well? Can be done by adding the following line to `.github/workflows/lint.yml`

```yaml
      - name: Lint with Flake8
        run: |
          pip install flake8
          flake8 --ignore=E501,E731 server
          flake8 --ignore=E501,E731 history
```

idreyn commented on server/routes.py:
```diff
@@ -0,0 +1,61 @@
+):
+    if normalized_route_name == "Red":
+        if custom_route_name == "Red-A":
+            return ["Braintree", "Alewife"]
```
😳

idreyn commented:
> Could we add the new `history/` directory to the linting check as well?

Great catch — after thinking about it for a bit I think it's better if we just enforce that all Python code lives in `server` by convention, and I moved the history code back into that directory. I also added `flake8` to the Pipfile dev deps and I imagine I'll also eventually set up some pre-push hooks or something like that so I don't drive myself crazy pushing up unlinted code.

nathan-weinberg reviewed (approved):
LGTM!
